Show using custom nonce logic (#742)
software-mansion/starknet.py: docs/guide/account_and_client.rst

    :dedent: 4

Custom nonce logic
------------------

By default, :ref:`Account` calls StarkNet for nonce every time a new transaction is signed or executed.
This is okay in for most users, but in case your applications needs to pre-sing multiple transactions

> for execution, deals with high amount of transactions or for whatever reason needs to support different nonce

Reviewer: ```suggestion
for execution, deals with high amount of transactions or just needs to support different nonce
```

> logic, it is possible to so with :ref:`Account. Simply overwrite the

Reviewer: ```suggestion
logic, it is possible to so with :ref:`Account`. Simply overwrite the
```

> :meth:`~starknet_py.net.account.account.Account.get_nonce` with your own logic.

Reviewer: ```suggestion
:meth:`~starknet_py.net.account.account.Account.get_nonce` method with your own logic.
```

.. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_custom_nonce.py
    :language: python

>     :dedent: 4

Reviewer: Maybe we should put it at the end of this section? I feel like it's not that common to use use it, and it is the second subsection in the whole guide.
Or we could put it somewhere in `signing` section maybe?

Author: We can move it to the end.

.. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_handling_client_errors.py
    :language: python
    :dedent: 4

Custom nonce logic
------------------

By default, :ref:`Account` calls StarkNet for nonce every time a new transaction is signed or executed.
This is okay for most users, but in case your applications needs to pre-sign multiple transactions
for execution, deals with high amount of transactions or just needs to support different nonce
logic, it is possible to so with :ref:`Account`. Simply overwrite the
:meth:`~starknet_py.net.account.account.Account.get_nonce` method with your own logic.

> 

Reviewer: Should we write here that the code below is just an example implementation of a nonce counter?
